spark/src/main/scala/geotrellis/spark/io/avro/codecs/TemporalProjectedExtentCodec.scala

```diff
@@ -33,20 +33,34 @@ trait TemporalProjectedExtentCodec {
       .record("TemporalProjectedExtent").namespace("geotrellis.spark")
       .fields()
       .name("extent").`type`(extentCodec.schema).noDefault()
-      .name("epsg").`type`().intType().noDefault()
+      .name("epsg").`type`().optional().intType()
+      .name("proj4").`type`().optional().stringType()
       .name("instant").`type`().longType().noDefault()
       .endRecord()
 
     def encode(temporalProjectedExtent: TemporalProjectedExtent, rec: GenericRecord): Unit = {
+      val crs = temporalProjectedExtent.crs
       rec.put("extent", extentCodec.encode(temporalProjectedExtent.extent))
-      rec.put("epsg", temporalProjectedExtent.crs.epsgCode.get)
+      if(crs.epsgCode.isDefined) {
+        rec.put("epsg", crs.epsgCode.get)
+      }
+      else {
+        rec.put("proj4", crs.toProj4String)
+      }
```

Comment:
Another (breaking) option would be to replace `epsg` all together with `crs`, and prefix EPSG codes according [to this](https://github.com/locationtech/geotrellis/blob/master/proj4/src/main/scala/geotrellis/proj4/CRS.scala#L120), falling back to "proj4" strings otherwise.